app/services/bills_actions_service.rb

```diff
@@ -0,0 +1,147 @@
+class BillsActionsService
+  module ActionTypes
+    INTRODUCED = %w(
+      bill:filed
+      bill:introduced
+      committee:referred
+      governor:received
+    )
+
+    PASSED = %w(
+      committee:passed
+      committee:passed:favorable
+      committee:passed:unfavorable
+      bill:passed
+      governor:signed
+      bill:veto_override:passed
+    )
+    
+    FAILED = %w(
+      committee:failed
+      bill:failed
+      governor:vetoed
+      governor:vetoed:line-item
+      bill:veto_override:failed
+    )
+
+    COMPLETED = [*PASSED, *FAILED]
+  end
+
+  module StageTypes
+    COMMITTEE = %w(
+      committee:referred
+      committee:passed
+      committee:passed:favorable
```

Comment:
what's the difference as far as the application is concerned right now between passed, passed:favorable, and passed:unfavorable? do we need to handle them differently?

Reply:
No clear answer on the difference from a legislative perspective.  We treat them all the same.  They just get boiled down into passed.